### PR TITLE
lua: put interpreter behind a compile option

### PR DESCRIPTION
**Pull request by Jan200101**
Only build the lua interpreter (lua, luac) if needed.

If a project only depends on Lua as a library for scripting then there is no reason to also build the interpreter

### CODE REVIEW

File: subprojects/packagefiles/lua/meson_options.txt [hunk 1]
```diff
@@ -6,3 +6,7 @@ option(
   'line_editing', type: 'boolean', value: false,
   description: 'Link with GNU readline (or libedit) to add history to the Lua interpreter.'
 )
+option(
+  'interpreter', type: 'boolean', value: true,
+  description: 'Build the Lua interpreter.'
+)
```
Comment (eli-schwartz) [inline subprojects/packagefiles/lua/meson_options.txt:12]: Looks like this has a bad line ending.

Comment (Jan200101) [reply subprojects/packagefiles/lua/meson_options.txt]: fixed, no idea how this got past me

**Comment by eli-schwartz** [comment]
Seems reasonable, and behavior-preserving by default... :)

**Comment by eli-schwartz** [comment]
In order to do a release of the changes, you need to bump the revision in releases.json though.

**Comment by Jan200101** [comment]
right, apologies for rushing things

**Comment by eli-schwartz** [comment]
Argh, Windows...

```
[36/36] Linking target subprojects/lua-5.4.3/lua.exe
FAILED: subprojects/lua-5.4.3/lua.exe subprojects/lua-5.4.3/lua.pdb 
"link"  /MACHINE:x64 /OUT:subprojects/lua-5.4.3/lua.exe subprojects/lua-5.4.3/lua.exe.p/src_lua.c.obj "/nologo" "/release" "/nologo" "/DEBUG" "/PDB:subprojects\lua-5.4.3\lua.pdb" "subprojects\lua-5.4.3\lua.lib" "/SUBSYSTEM:CONSOLE" "kernel32.lib" "user32.lib" "gdi32.lib" "winspool.lib" "shell32.lib" "ole32.lib" "oleaut32.lib" "uuid.lib" "comdlg32.lib" "advapi32.lib"
LINK : fatal error LNK1181: cannot open input file 'subprojects\lua-5.4.3\lua.lib'
ninja: build stopped: subcommand failed.
```

**Comment by Jan200101** [comment]
love it when it does that.
I'll see what I can do when I get a hold of a Windows machine

**Comment by eli-schwartz** [comment]
The good news is that it doesn't seem to actually be your fault. The last update to this wrap was before we added Windows CI, and the build_all regression tests marked lua as known-broken now that I look at it (ci_config.json).

**Comment by Jan200101** [comment]
Tested against MinGW64 and came across no issue so this seems to be caused by Visual Studio

**Comment by Jan200101** [comment]
Okay I cannot figure out whats going wrong.
This CI issue existed even before this PR so its not something new.

Essentially, the `executable("lua")` is expecting the previously defined `library("lua")` to be static for whatever reason
I am not well versed with meson but I have essentially tried to strip the build.meson as far down as I could without getting anything to work.

Is there anything I am missing?

**Comment by Jan200101** [comment]
I have taken the aproach of creating a internal static library to link the interpreter.
That solves the issue and works

**Comment by eli-schwartz** [comment]
I suppose that seems fine. In theory one could link to the static library only on MSVC, and link to the default library otherwise.

**Comment by eli-schwartz** [comment]
⚔️ conflicts

**Comment by Jan200101** [comment]
After some digging this seems to be expected
https://github.com/mesonbuild/meson/blob/master/docs/markdown/FAQ.md#my-project-works-fine-on-linux-and-mingw-but-fails-to-link-with-msvc-due-to-a-missing-lib-file-fatal-error-lnk1181-why

It seems that lua does not support MSVC at all so compiling them statically if msvc is detected is the best way.

**Comment by eli-schwartz** [comment]
Cool, thanks for figuring this out!

**Pull request merged**